code/exercises/affine-cipher/example.scm

```diff
@@ -0,0 +1,89 @@
+(import (rnrs))
+
+
+(define (encode key text)
+  ((get-translator key) (string-downcase text) 'encode))
+
+(define (decode key text)
+  ((get-translator key) text 'decode))
+
+;;; Private
+
+(define (atbash-format lst)
+  (let* ((chunks (windows lst 5))
+         (l (cons (car chunks)
+                  (map (lambda (chars) (cons #\space chars))
+                       (cdr chunks)))))
+    (list->string (fold-left append '() l))))
+
+(define (get-translator key)
+   (translator-factory
+    (cdr (assoc 'a key))
+    (cdr (assoc 'b key))))
+
+;; Work Horse - return a translator tuned to a specific affine key
+(define (translator-factory a b)
+  (when (not (coprime? a 26))
+    (error 'affine-cipher "a and m must be coprime" a))
```

Comment:
I actually think that an `Either` should be returned by the wrapper, so this function doesn't deal with errors, and so the user of the translators could try again if the key is invalid, but the exercise isn't set up that way, so I just put it here since it was most convenient to check the `a` value at this point.